Assets/Scripts/Algorithms/Planning/LaunchAnglePlanner.cs

```diff
@@ -42,22 +46,58 @@ public LaunchAngleDataPoint(in LaunchAngleInput input, in LaunchAngleOutput outp
   }
 }
 
-// The launch angle planner class is an interface for a planner that outputs the optimal launch
-// angle and the time-to-target.
+// The launch angle planner interface defines methods for calculating optimal launch angles
+// and intercept positions.
 public interface ILaunchAnglePlanner {
-  // Calculate the optimal launch angle in degrees and the time-to-target in seconds.
+  // Calculate the optimal launch angle and time-to-target for a given input.
+  //   input: Launch angle input parameters
+  // Returns: Launch angle output with timing information
   public LaunchAngleOutput Plan(in LaunchAngleInput input);
+
+  // Calculate launch parameters for a target position from the default origin (0,0,0).
+  // Maintained for backward compatibility.
+  //   position: Target position
+  // Returns: Launch angle output
   public LaunchAngleOutput Plan(Vector3 position) {
-    Vector2 direction = ConvertToDirection(position);
-    return Plan(new LaunchAngleInput(distance: direction[0], altitude: direction[1]));
+    return Plan(position, Vector3.zero);
+  }
+
+  // Calculate launch parameters for a target position from a specific origin.
+  // This method accounts for the distance and direction from origin to target.
+  //   targetPosition: Target position
+  //   originPosition: Interceptor origin position
+  // Returns: Launch angle output with origin-relative calculations
+  public LaunchAngleOutput Plan(Vector3 targetPosition, Vector3 originPosition) {
+    Vector2 direction = ConvertToDirection(targetPosition, originPosition);
+    float distance = Vector3.Distance(originPosition, targetPosition);
+    var output = Plan(new LaunchAngleInput(distance: direction[0], altitude: direction[1]));
+    return new LaunchAngleOutput(output.LaunchAngle, output.TimeToPosition, distance);
+  }
+
+  // Get the intercept position for a target from the default origin (0,0,0).
+  // Maintained for backward compatibility.
```

Comment:
ditto.  On second thought, I would even remove all interface functions you don't want to use in the future